modules/imgproc/src/subdivision2d.cpp

```diff
@@ -210,19 +210,130 @@ void Subdiv2D::swapEdges( int edge )
     splice(sedge, getEdge(b, NEXT_AROUND_LEFT));
 }
 
-static double triangleArea( Point2f a, Point2f b, Point2f c )
-{
-    return ((double)b.x - a.x) * ((double)c.y - a.y) - ((double)b.y - a.y) * ((double)c.x - a.x);
+static double CrossProduct(Point2f u, Point2f v) {
+    return (double)u.x * v.y - (double)v.x * u.y;
 }
 
-int Subdiv2D::isRightOf(Point2f pt, int edge) const
-{
-    Point2f org, dst;
-    edgeOrg(edge, &org);
-    edgeDst(edge, &dst);
-    double cw_area = triangleArea( pt, dst, org );
+static double DotProduct(Point2f u, Point2f v) {
```

Comment:
See `.dot()` method: https://docs.opencv.org/master/db/d4e/classcv_1_1Point__.html#a4728ce98be0240d836c4896021688aa5

Reply:
Thanks, I agree this already exists.